src/bin/cargo/main.rs

```diff
@@ -35,16 +35,43 @@ fn main() {
     }
 }
 
-fn setup_logger() {
-    let env = tracing_subscriber::EnvFilter::from_env("CARGO_LOG");
+fn setup_logger() -> Option<tracing_chrome::FlushGuard> {
+    #![allow(clippy::disallowed_methods)]
+
+    use tracing_subscriber::prelude::*;
 
-    tracing_subscriber::fmt()
+    let env = tracing_subscriber::EnvFilter::from_env("CARGO_LOG");
+    let fmt_layer = tracing_subscriber::fmt::layer()
         .with_timer(tracing_subscriber::fmt::time::Uptime::default())
         .with_ansi(std::io::IsTerminal::is_terminal(&std::io::stderr()))
         .with_writer(std::io::stderr)
-        .with_env_filter(env)
-        .init();
+        .with_filter(env);
+
+    let (profile_layer, profile_guard) =
+        if env_to_bool(std::env::var_os("_CARGO_LOG_PROFILE").as_deref()) {
+            let capture_args =
+                env_to_bool(std::env::var_os("_CARGO_LOG_PROFILE_CAPTURE_ARGS").as_deref());
```

Comment:
Idea: use the same env var and accept different values.

Reply:
I was modeling this off of `git-branchless` where their equivalent of `_CARGO_LOG_PROFILE` accepted file names. I just held off on that for now.  Being able to set the file name is useful for predictability and for when nesting processes in each other.

We also have the possibility of doing logger-like stuff of filtering what gets captured.  This is remind me that profiling is just one role for this and this is also a programmatic format for logged output that allows graphical browsing of the relationship of events.

Unsure if that changes your thoughts here on merging them.

Reply:
Given we claim it as perma-unstable, everything sounds good to me 👍🏾.